trafilatura/utils.py

```diff
@@ -270,6 +271,8 @@ def line_processing(line):
     # remove newlines that are not related to punctuation or markup
     # remove non-printable chars and normalize space characters (including Unicode spaces)
     line = trim(remove_control_characters(LINES_TRIMMING.sub(r' ', line)))
+    # replace unique code spaces with regular space
+    line = line.replace(CODE_SPACE, ' ')
```

Comment:
This function is critical for performance, is this line strictly necessary? I saw that you handle code spaces differently in `handle_code_blocks`.

Reply:
The problem I was trying to solve was to keep spaces within code blocks. The current version strips multiple spaces and these spaces are crucial for code (e.g indentation in python). I don't like my solution, but I wasn't able to come up with something else. Within code blocks, I replace spaces with something else and at the end return it back to spaces, this way I get around the stripping of spaces. Happy to adjust if you can recommend another way.

Reply:
I try to fix it by appending it just above.